File: Count Number of teams.cpp
class Solution {
public:
    int numTeams(vector<int>& rating) {
        int n = rating.size();
        int count = 0;
        for (int mid = 1; mid < n - 1; ++mid) {
            int leftSmallerCount = 0, rightGreaterCount = 0;
            for (int i = 0; i < mid; ++i) {
                if (rating[i] < rating[mid]) {
                    leftSmallerCount++;
                }
            }
            for (int i = mid + 1; i < n; ++i) {
                if (rating[i] > rating[mid]) {
                    rightGreaterCount++;
                }
            }
            // increasing order
            count += leftSmallerCount * rightGreaterCount;
            
            int rightSmallerCount = 0, leftGreaterCount = 0;
            for (int i = mid + 1; i < n; ++i) {
                if (rating[i] < rating[mid]) {
                    rightSmallerCount++;
                }
            }
            for (int i = 0; i < mid; ++i) {
                if (rating[i] > rating[mid]) {
                    leftGreaterCount++;
                }
            }
            // decreasing order
            count += rightSmallerCount * leftGreaterCount;
        }
        return count;
    }
};

File: Count number of teams.py
class Solution:
    def numTeams(self, rating):
        n = len(rating)
        count = 0
        for mid in range(1, n - 1):
            leftSmallerCount = sum(1 for i in range(mid) if rating[i] < rating[mid])
            rightGreaterCount = sum(1 for i in range(mid + 1, n) if rating[i] > rating[mid])
            
            # increasing order
            count += leftSmallerCount * rightGreaterCount
            
            rightSmallerCount = sum(1 for i in range(mid + 1, n) if rating[i] < rating[mid])
            leftGreaterCount = sum(1 for i in range(mid) if rating[i] > rating[mid])
            
            # decreasing order
            count += rightSmallerCount * leftGreaterCount
            
        return count
